src/test/objectstore/chain_xattr.cc

```diff
@@ -230,13 +230,32 @@ TEST(chain_xattr, listxattr) {
   const string x(LARGE_BLOCK_LEN, 'X');
   const int y = 1234;
 
+  int orig_size = chain_listxattr(file, NULL, 0);
+  char *orig_buffer = NULL;
+  string orig_str;
+  if (orig_size) {
+    orig_buffer = (char*)malloc(orig_size);
+    chain_flistxattr(fd, orig_buffer, orig_size);
+    orig_str = string(orig_buffer);
+    orig_size = orig_str.size();
+  }
```

Comment:
agreed. it's difficult to *fully* understand *all* of them. but if somethings goes wrong, guess we can also try to dive into it to figure out the root cause.